src/java.base/share/classes/java/lang/ref/Reference.java

```diff
@@ -388,11 +388,22 @@ boolean refersToImpl(T obj) {
     private native boolean refersTo0(Object o);
 
     /**
-     * Clears this reference object.  Invoking this method will not cause this
-     * object to be enqueued.
+     * Clears this reference object. Invoking this method does not enqueue this
+     * object, and the garbage collector will no longer enqueue this object once
+     * the referent reaches the designated reachability level.
+     * <p>
+     * There is a potential race condition with the garbage collector. When this
+     * method is called, the garbage collector may already be in the process of
+     * (or already completed) clearing and/or enqueueing this reference.
+     *
+     * Avoid this race by ensuring the referent remains strongly-reachable until
+     * after the call to clear(), using {@link #reachabilityFence(Object)} if necessary.
+     *
```

Comment:
The text about the potential race condition and the recommendation to use reachabilityFence() should be in an apiNote at the end of the method spec, after the following paragraph.

Reply:
yes I think the same.  It should be an apiNote.